src/Features/BannedSymbols.txt

```diff
@@ -11,3 +11,4 @@ T:System.ComponentModel.Composition.PartNotDiscoverableAttribute; Use types from
 T:System.ComponentModel.Composition.SharedAttribute; Use types from System.Composition instead
 T:System.ComponentModel.Composition.SharingBoundaryAttribute; Use types from System.Composition instead
 T:System.ComponentModel.Composition.Convention.AttributedModelProvider; Use types from System.Composition instead
+T:Microsoft.CodeAnalysis.Options.IGlobalOptionService; Pass required options or options provider to each feature/service explicitly
```

Comment:
💭 It might be easier to just mark the type with `[Experimental(nameof(IGlobalOptionService))]`, which would require a _named_ warning suppression and not just RS0030. This could safely be applied to an entire file by using this at the top of a file:

```
#pragma warning disable IGlobalOptionService
```

Reply:
Good idea. Will do.

Reply:
Works very well. Thanks for the suggestion!